les/api_backend.go

```diff
@@ -331,3 +332,13 @@ func (b *LesApiBackend) StateAtBlock(ctx context.Context, block *types.Block, re
 func (b *LesApiBackend) StateAtTransaction(ctx context.Context, block *types.Block, txIndex int, reexec uint64) (core.Message, vm.BlockContext, *state.StateDB, error) {
 	return b.eth.stateAtTransaction(ctx, block, txIndex, reexec)
 }
+
+func (b *LesApiBackend) SetHistoricalBlocksSynced() bool {
+	if b.eth.handler.downloader.Progress().CurrentBlock < b.eth.handler.downloader.Progress().HighestBlock {
+		log.Info("Light Client not fully synced yet and block specimen producer does not operate in light client mode")
+		return false
+	} else {
+		log.Info("Fully Synced but block specimen producer does not operate in light client mode", "Current block", b.eth.handler.downloader.Progress().CurrentBlock, "Highest block", b.eth.handler.downloader.Progress().HighestBlock)
```

Comment:
in fact we should probably fatal out here given light client mode is not supported.

Reply:
yes agreed this should fatal out

Reply:
moving to `log.Crit`